src/pwr.rs

```diff
@@ -86,7 +86,7 @@ pub struct Pwr {
 /// Generated when the PWR peripheral is frozen. The existence of this
 /// value indicates that the voltage scaling configuration can no
 /// longer be changed.
-#[derive(PartialEq)]
```

Comment:
Currently only one `VoltageScale` is created which is then passed to `Rcc::freeze()` and never seen again. It doesn't seem to harm anything to make this `Copy`, though.

Reply:
The mechanics of `VoltageScale` could do with some more work. This idea is that the PWR HAL sets the voltage scale, and the RCC HAL uses this to configure the core and flash correctly for that voltage scale. If PWR returned `VoltageScale::Scale3` but then you passed `VoltageScale::Scale1` to `Rcc::freeze()` that would be bad - likely you'd read corrupt instructions from flash and hardfault.

Currently we allow this, but perhaps we should prevent it. One solution is to declare `VoltageScale` as a struct with  private members (and that does not implement Clone/Copy). However I'd left this as future work - for example there's no functions for changing voltage scale dynamically yet.

Unless you want to work on this, I would leave the current declaration as-is.

Reply:
I'll leave that to another PR since it's a breaking change.